core/src/main/java/org/kiwix/kiwixmobile/core/utils/LanguageUtils.kt

```diff
@@ -185,28 +186,29 @@ class LanguageUtils(private val context: Context) {
 
     @JvmStatic
     fun handleLocaleChange(
-      context: Context,
+      activity: Activity,
       sharedPreferenceUtil: SharedPreferenceUtil
     ) {
       sharedPreferenceUtil.prefLanguage.takeIf { it != Locale.ROOT.toString() }?.let {
-        handleLocaleChange(context, it)
+        handleLocaleChange(activity, it)
       }
     }
 
     @SuppressLint("AppBundleLocaleChanges")
     @JvmStatic
-    fun handleLocaleChange(context: Context, language: String) {
+    fun handleLocaleChange(activity: Activity, language: String) {
       val locale =
         if (language == Locale.ROOT.toString())
-          ConfigurationCompat.getLocales(context.applicationContext.resources.configuration)[0]
+          ConfigurationCompat.getLocales(activity.applicationContext.resources.configuration)[0]
         else
           Locale(language)
       Locale.setDefault(locale)
       val config = Configuration()
       config.setLocale(locale)
       config.setLayoutDirection(locale)
-      context.resources
-        .updateConfiguration(config, context.resources.displayMetrics)
+      activity.resources
+        .updateConfiguration(config, activity.resources.displayMetrics)
```

Comment:
This is the same `updateConfiguration` we are accessing.

Reply:
hi @gouri-panda ,

yes this is same but i have added a method which is activity method ```activity.onConfigurationChanged(config)``` and we can't access it with ```context```.